common/metrics/opentelemetry_reporter.go

```diff
@@ -33,71 +33,70 @@ import (
 
 var _ Reporter = (OpentelemetryReporter)(nil)
 var _ OpentelemetryReporter = (*opentelemetryReporterImpl)(nil)
-var _ OpentelemetryMustProvider = (*opentelemetryMustProviderImpl)(nil)
+var _ OpentelemetryProvider = (*opentelemetryProviderImpl)(nil)
 
 type (
 	OpentelemetryReporter interface {
 		NewClient(logger log.Logger, serviceIdx ServiceIdx) (Client, error)
 		Stop(logger log.Logger)
-		GetMeterMust() metric.MeterMust
+		GetMeter() metric.Meter
 		UserScope() UserScope
 	}
 
 	// opentelemetryReporterImpl is a base class for reporting metrics to opentelemetry.
 	opentelemetryReporterImpl struct {
 		exporter     *prometheus.Exporter
 		meter        metric.Meter
-		meterMust    metric.MeterMust
 		clientConfig *ClientConfig
 		gaugeCache   OtelGaugeCache
 		userScope    UserScope
-		mustProvider OpentelemetryMustProvider
+		otelProvider OpentelemetryProvider
 	}
 
 	OpentelemetryListener struct {
 	}
 )
 
-func NewOpentelemeteryReporterWithMust(
```

Comment:
I assume this function is not intended for open source use and ok to rename it? OSS users should use the `NewOpentelemeteryReporter` function below to create a custom reporter?

Reply:
@yiminc Mind help confirm this is the case before I land this PR?

Reply:
👍 , otel metrics API is not stable anyways.